### PR TITLE
Fixes ComposerView going behind keyboard when tabbar is used

### PR DESCRIPTION
**Pull request by b-onc**
I've tested the fix with iOS11.4, 12.2, 12.4 and 13.3. All work good. I'm opening a PR but I'll be brainstorming a bit on how to write a UI test on this, so this is WiP for now.

### CODE REVIEW

**Comment by buh** [comment]
@b-onc Can you check how it works with `ChatViwController` in UIContainerView?

**Comment by b-onc** [comment]
@buh I've tested 3 cases, here are the stacks:
- TabBarController -> NavigationController -> ChannelsViewController (pushes ChatViewController)
- TabBarController -> NavigationController -> CustomChannelsViewController (which has chatViewController.hidesBottomBarWhenPushed = false)
- TabBarController -> NavigationController -> ChatViewController

Are there any other cases I need to check?

File: Sources/UI/Chat/Chat View Controller/ChatViewController.swift [hunk 1]
```diff
@@ -51,14 +51,13 @@ open class ChatViewController: ViewController, UITableViewDataSource, UITableVie
...
     /// Calculates the bottom inset for the `ComposerView` when the keyboard will appear.
     open var calculatedSafeAreaBottom: CGFloat {
-        var bottom = view.safeAreaInsets.bottom
-        bottom = bottom > 0 ? bottom : (parent?.view.safeAreaInsets.bottom ?? 0)
-        
-        if let tabBar = tabBarController?.tabBar, !tabBar.isTranslucent {
-            bottom += tabBar.frame.height
+        if let tabBar = tabBarController?.tabBar, !tabBar.isTranslucent, !hidesBottomBarWhenPushed {
+            return tabBar.frame.height
+        } else {
```
Comment (buh) [inline Sources/UI/Chat/Chat View Controller/ChatViewController.swift]: No need `else` here, was `return` on the previous line.

Comment (b-onc) [reply Sources/UI/Chat/Chat View Controller/ChatViewController.swift]: We don't need to access `view.safeAreaInsets` if the view has `tabBarController` so I've moved the logic inside `else`

Comment (buh) [reply Sources/UI/Chat/Chat View Controller/ChatViewController.swift]:
```swift
open var calculatedSafeAreaBottom: CGFloat {
    if let tabBar = tabBarController?.tabBar, !tabBar.isTranslucent, !hidesBottomBarWhenPushed {
        return tabBar.frame.height
    }

    var bottom = view.safeAreaInsets.bottom
    bottom = bottom > 0 ? bottom : (parent?.view.safeAreaInsets.bottom ?? 0)
    return bottom
}
```
Remove `else`

Comment (buh) [reply Sources/UI/Chat/Chat View Controller/ChatViewController.swift]: and no need `bottom` as `var`

**Comment by b-onc** [comment]
OK, I'm closing this PR without merging and will open a new one on top of 1.5.4

**Pull request closed without merging**